engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/deployment/DeploymentPreparer.scala

```diff
@@ -0,0 +1,87 @@
+package pl.touk.nussknacker.k8s.manager.deployment
+
+import io.circe.syntax.EncoderOps
+import monocle.Iso
+import monocle.macros.GenLens
+import monocle.std.option._
+import pl.touk.nussknacker.engine.api.ProcessVersion
+import pl.touk.nussknacker.k8s.manager.K8sDeploymentManager.{labelsForScenario, objectNameForScenario, scenarioIdLabel, scenarioVersionAnnotation}
+import pl.touk.nussknacker.k8s.manager.K8sDeploymentManagerConfig
+import skuber.EnvVar.FieldRef
+import skuber.LabelSelector.IsEqualRequirement
+import skuber.apps.v1.Deployment
+import skuber.{Container, EnvVar, HTTPGetAction, LabelSelector, Pod, Probe, Volume}
+
+object DeploymentPreparer {
+
+  def apply(processVersion: ProcessVersion, config: K8sDeploymentManagerConfig, configMapId: String): Deployment = {
+    val objectName = objectNameForScenario(processVersion, None)
+    val annotations = Map(scenarioVersionAnnotation -> processVersion.asJson.spaces2)
+    val labels = labelsForScenario(processVersion)
+    val k8sDeploymentConfig: Deployment = DeploymentUtils.createDeployment(config.k8sDeploymentConfig)
+
+    val deploymentSpecLens = GenLens[Deployment](_.spec) composePrism some
+    val templateSpecLens = deploymentSpecLens composeLens GenLens[Deployment.Spec](_.template.spec) composeIso withDefault(Pod.Spec())
+    val deploymentLens =
+      GenLens[Deployment](_.metadata.name).set(objectName) andThen
+        GenLens[Deployment](_.metadata.labels).modify(_ ++ labels) andThen
+        GenLens[Deployment](_.metadata.annotations).modify(_ ++ annotations) andThen
+        (deploymentSpecLens composeLens GenLens[Deployment.Spec](_.minReadySeconds)).modify(f => setMinReadySeconds(f)) andThen
+        //here we use id to avoid sanitization problems
+        (deploymentSpecLens composeLens GenLens[Deployment.Spec](_.selector)).set(LabelSelector(IsEqualRequirement(scenarioIdLabel, processVersion.processId.value.toString))) andThen
+        (deploymentSpecLens composeLens GenLens[Deployment.Spec](_.strategy)).modify(maybeStrategy => maybeStrategy.orElse(Some(Deployment.Strategy.Recreate))) andThen
```

Comment:
maybe it would be easier to put it into deploymentMinimal.json? and call it deploymentDefaults or sth?